component/otelcol/exporter/prometheus/internal/convert/convert_test.go

```diff
@@ -838,6 +839,233 @@ func TestConverter(t *testing.T) {
 			addMetricSuffixes: true,
 			enableOpenMetrics: true,
 		},
+		{
+			name: "Gauge: convert resource attributes to metric label",
+			input: `{
+				"resource_metrics": [{
+					"resource": {
+						"attributes": [{
+							"key": "service.name",
+							"value": { "stringValue": "myservice" }
+						}, {
+							"key": "service.instance.id",
+							"value": { "stringValue": "instance" }
+						}, {
+							"key": "raw",
+							"value": { "stringValue": "test" }
+						},{
+							"key": "foo.one",
+							"value": { "stringValue": "foo" }
+						}, {
+							"key": "bar.one",
+							"value": { "stringValue": "bar" }
+						}]
+					},
+					"scope_metrics": [{
+						"metrics": [{
+							"name": "test_metric_gauge",
+							"gauge": {
+								"data_points": [{
+									"as_double": 1234.56
+								}]
+							}
+						}]
+					}]
+				}]
+			}`,
+			expect: `
+				# TYPE test_metric_gauge gauge
+				test_metric_gauge{bar_one="bar",foo_one="foo",instance="instance",service_instance_id="instance",job="myservice",service_name="myservice",raw="test"} 1234.56
+			`,
+			enableOpenMetrics:             true,
+			resourceToTelemetryConversion: true,
```

Comment:
It'd be nice to have a test or two where we have resource attributes but `resourceToTelemetryConversion: false`

Reply:
Added in latest commit. PTAL. Thanks!